packages/twenty-server/src/workspace/workspace-sync-metadata/utils/create-deterministic-uuid.util.ts

```diff
@@ -0,0 +1,10 @@
+import { createHash } from 'crypto';
+
+export const createDeterministicUuid = (inputUuid: string): string => {
+  const hash = createHash('sha256').update(inputUuid).digest('hex');
+
+  return `20202020-4${hash.substring(0, 3)}-${hash.substring(
```

Comment:
I'm not sure why you need to add the 4 and 8 there? is it to avoid potential collisions?

Reply:
Actually uuid regex looks like this:

```typescript
export default /^(?:[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}|00000000-0000-0000-0000-000000000000)$/i;
```

Extracted from [here](https://github.com/uuidjs/uuid/blob/main/src/regex.js)

For the 8 we need it, as the fourth group should start by `[89ab]`.
But actually I put the 4 in the wrong group as it should be on the third one `[1-5]`.

Reply:
@magrinj Interesting, I didn't know that 🙂